pkg/querier/queryrange/querysharding.go

```diff
@@ -55,19 +52,92 @@ func NewQueryShardMiddleware(
 
 }
 
-func newASTMapperware(
+func newSimpleShardingware(
 	confs queryrange.ShardingConfigs,
-	next queryrange.Handler,
 	logger log.Logger,
 	metrics *logql.ShardingMetrics,
-) *astMapperware {
+	codec queryrange.Codec,
+) queryrange.Middleware {
+	return queryrange.MiddlewareFunc(func(next queryrange.Handler) queryrange.Handler {
+		return &simpleShardingWare{
+			confs:  confs,
+			logger: log.With(logger, "middleware", "QueryShard.simpleShardingWare"),
+			next:   next,
+			codec:  codec,
+		}
+	})
+}
+
+// simpleShardingWare does not do any special AST mapping, it just fans out sharded requests downstream.
+// This is used for /series queries.
+type simpleShardingWare struct {
+	confs  queryrange.ShardingConfigs
+	logger log.Logger
+	next   queryrange.Handler
+	codec  queryrange.Codec
+}
+
+func (m *simpleShardingWare) Do(ctx context.Context, r queryrange.Request) (queryrange.Response, error) {
+	conf, err := m.confs.GetConf(r)
+	// cannot shard with this timerange
+	if err != nil {
+		level.Warn(m.logger).Log("err", err.Error(), "msg", "skipped simpleShardingWare for request")
+		return m.next.Do(ctx, r)
+	}
+
+	shardedLog, ctx := spanlogger.New(ctx, "simpleShardingWare")
+	defer shardedLog.Finish()
+
+	req, ok := r.(*LokiRequest)
```

Comment:
I'm confused about this I think in case of the Series API the request is a SeriesRequest not a LokiRequest no ?

 This also means you need to make the http querier shardaware ?

see https://github.com/grafana/loki/blob/master/pkg/querier/queryrange/codec.go#L179

I have a feeling this doesn;t work for the series API.

Reply:
I think you're probably right 🤢